File: emulinker/src/main/java/org/emulinker/kaillera/model/impl/AutoFireDetectorFactoryImpl.java
package org.emulinker.kaillera.model.impl;

import javax.inject.Inject;
import org.emulinker.kaillera.model.*;

public final class AutoFireDetectorFactoryImpl implements AutoFireDetectorFactory {

  @Inject
  AutoFireDetectorFactoryImpl() {}

  @Override
  public AutoFireDetector getInstance(KailleraGame game, int defaultSensitivity) {
    return new AutoFireScanner2(game, defaultSensitivity);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/impl/AutoFireDetectorFactoryImpl.kt
package org.emulinker.kaillera.model.impl

import javax.inject.Inject
import org.emulinker.kaillera.model.KailleraGame

class AutoFireDetectorFactoryImpl @Inject internal constructor() : AutoFireDetectorFactory {
  override fun getInstance(game: KailleraGame, defaultSensitivity: Int): AutoFireDetector {
    return AutoFireScanner2(game, defaultSensitivity)
  }
}
